src/frontend/src/common/types/network.d.ts

```diff
@@ -1,4 +1,4 @@
-type JSONPrimitive = string | number | boolean;
+type JSONPrimitive = string | number | boolean | Record<string, unknown>;
```

Comment:
hmm, it's not really a primitive any more is it? why modify this type instead of changing the type of the variable to `JSONPrimitive | Record<string, unknown>`?

Reply:
That's true! I'm thinking about changing `JSONPrimitive` to just `JSONValue` and expand it to:

```ts
type JSONValue = 
 | string
 | number
 | boolean
 | null
 | JSONValue[]
 | {[key: string]: JSONValue}
```

Since that's all the valid JSON types we can get

source: https://github.com/microsoft/TypeScript/issues/1897#issuecomment-822032151

What do you think?

Reply:
I think that's a good idea but it should be a separate type, since I am pretty sure I relied on constraining variable to just primitives where specified in the code.